Failed on multi-asic on unrelated failure from syslog testcase:


```
>               pt_assert(False, failure_message)
E               Failed: Got matched syslog in processes "analyze_logs--<MultiAsicSonicHost vlab-08>" exit code:"1"
E               match: 1
E               expected_match: 0
E               expected_missing_match: 0
E               
E               Match Messages:
E               2026 Apr 24 17:15:49.972709 vlab-08 ERR swss0#rsyslogd: omrelp[240.127.1.1:2514]: error 'error opening connection to remote peer', object  'conn to srvr 240.127.1.1:2514' - action may not work as intended [v8.2302.0 try https://www.rsyslog.com/e/2353 ]

alive      = []
args       = [{'vlab-08': <tests.common.plugins.loganalyzer.loganalyzer.LogAnalyzer object at 0x790504e04e60>}, {'vlab-08': 'test_syslog[fd82:b34f:cc99::100-7.0.80.166].2026-04-24-17:16:16'}]
concurrent_tasks = 24
delta_time = datetime.timedelta(seconds=9, microseconds=319417)
end_time   = datetime.datetime(2026, 4, 24, 17, 17, 16, 564456)
failed_processes = {'analyze_logs--<MultiAsicSonicHost vlab-08>': {'exception': ('Log Analyzer Error- Matches found, please check errors ...27.1.1:2514\' - action may not work as intended [v8.2302.0 try https://www.rsyslog.com/e/2353 ]\n\n'), 'exit_code': 1}}
failure_message = 'Got matched syslog in processes "analyze_logs--<MultiAsicSonicHost vlab-08>" exit code:"1"\nmatch: 1\nexpected_match:... \'conn to srvr 240.127.1.1:2514\' - action may not work as intended [v8.2302.0 try https://www.rsyslog.com/e/2353 ]\n'
```

--- a/tests/common/helpers/upgrade_helpers.py
+++ b/tests/common/helpers/upgrade_helpers.py
@@ -22,8 +22,9 @@ logger = logging.getLogger(__name__)
 @pytest.fixture(scope="module")
 def xcvr_skip_list(duthosts):
     """Return empty transceiver skip list per DUT. Shared across upgrade-related test suites."""
-    # Required by reboot_and_check(); empty list = no transceivers skipped during post-reboot health check.
+    # Required by reboot_and_check() empty list = no transceivers skipped during post-reboot health check.
     return {dut.hostname: [] for dut in duthosts}
+
 
 TMP_VLAN_PORTCHANNEL_FILE = '/tmp/portchannel_interfaces.json'
 TMP_VLAN_FILE = '/tmp/vlan_interfaces.json'
@@ -434,7 +435,10 @@ def _wait_gnoi_time_ready(ptf_gnoi, metadata, cfg: GnoiUpgradeConfig, timeout=No
     return wait_until(timeout, interval, 0, ptf_gnoi.system_time, metadata=metadata)
 
 
-def _upgrade_one_dpu_via_gnoi(duthost, tbinfo, ptf_gnoi, cfg: GnoiUpgradeConfig) -> Dict:
+def _upgrade_one_dpu_via_gnoi(
+    duthost, tbinfo, ptf_gnoi, cfg: GnoiUpgradeConfig,
+    localhost=None, duthosts=None,
+) -> Dict:
     """
     Upgrade a single DPU via gNOI.
 
@@ -473,36 +477,39 @@ def _upgrade_one_dpu_via_gnoi(duthost, tbinfo, ptf_gnoi, cfg: GnoiUpgradeConfig)
         metadata=md,
     )
 
-    # Steps 4 & 5 are skipped when skip_reboot=True (caller will trigger reboot separately)
     if cfg.skip_reboot:
         logger.info("skip_reboot=True: image staged on DPU, skipping reboot")
         return {"transfer_resp": transfer_resp, "setpkg_resp": setpkg_resp, "reboot_resp": None}
 
-    # Step 4: Reboot the DPU
-    method = str(cfg.upgrade_type).upper()
-    try:
-        reboot_resp = ptf_gnoi.system_reboot(
-            method=method,
-            delay=0,
-            message=cfg.ss_reboot_message,
-            metadata=md,
-        )
-    except Exception as e:
-        logger.info("Reboot raised (often expected after DPU disconnect): %s", e)
-        reboot_resp = None
+    from tests.common.ptf_gnoi import PtfGnoi
+    dpu_grpc = ptf_gnoi.grpc_client.with_ss_target("dpu", cfg.ss_target_index)
+    dpu_ptf_gnoi = PtfGnoi(dpu_grpc)
+
+    reboot_and_check(
+        localhost,
+        duthost,
+        {},
+        {},
+        reboot_type=cfg.upgrade_type,
+        duthosts=duthosts,
+        invocation_type="gnoi_based",
+        ptf_gnoi=dpu_ptf_gnoi,
+    )
 
     if cfg.allow_fail:
-        return {"transfer_resp": transfer_resp, "setpkg_resp": setpkg_resp, "reboot_resp": reboot_resp}
+        return {"transfer_resp": transfer_resp, "setpkg_resp": setpkg_resp}
 
-    # Step 5: Wait for DPU to come back up
-    ok = _wait_gnoi_time_ready(ptf_gnoi, md, cfg)
-    pytest_assert(ok, f"gNOI Time not reachable within {cfg.ss_reboot_ready_timeout}s after DPU reboot")
-
-    return {"transfer_resp": transfer_resp, "setpkg_resp": setpkg_resp, "reboot_resp": reboot_resp}
+    return {"transfer_resp": transfer_resp, "setpkg_resp": setpkg_resp}
 
 
-def perform_gnoi_upgrade_smartswitch_dpu(duthost, tbinfo, ptf_gnoi, cfg: GnoiUpgradeConfig) -> Dict:
-    return _upgrade_one_dpu_via_gnoi(duthost, tbinfo, ptf_gnoi, cfg)
+def perform_gnoi_upgrade_smartswitch_dpu(
+    duthost, tbinfo, ptf_gnoi, cfg: GnoiUpgradeConfig,
+    localhost=None, duthosts=None,
+) -> Dict:
+    return _upgrade_one_dpu_via_gnoi(
+        duthost, tbinfo, ptf_gnoi, cfg,
+        localhost=localhost, duthosts=duthosts,
+    )
 
 
 def perform_gnoi_upgrade_smartswitch_dpus_parallel(
@@ -510,6 +517,7 @@ def perform_gnoi_upgrade_smartswitch_dpus_parallel(
     ptf_gnoi,
     cfgs: Sequence[GnoiUpgradeConfig],
     max_workers: Optional[int] = None,
+    localhost=None, duthosts=None,
 ) -> Dict[int, Dict]:
     if not cfgs:
         raise ValueError("cfgs is empty")
@@ -520,7 +528,10 @@ def perform_gnoi_upgrade_smartswitch_dpus_parallel(
     with SafeThreadPoolExecutor(max_workers=workers) as executor:
         futs = {}
         for i, cfg in enumerate(cfgs):
-            futs[i] = executor.submit(_upgrade_one_dpu_via_gnoi, duthost, tbinfo, ptf_gnoi, cfg)
+            futs[i] = executor.submit(
+                _upgrade_one_dpu_via_gnoi, duthost, tbinfo, ptf_gnoi, cfg,
+                localhost, duthosts,
+            )
 
         for i, fut in futs.items():
             results[i] = fut.get()

--- a/tests/common/helpers/upgrade_helpers.py
+++ b/tests/common/helpers/upgrade_helpers.py
@@ -22,8 +22,8 @@ logger = logging.getLogger(__name__)
 @pytest.fixture(scope="module")
 def xcvr_skip_list(duthosts):
     """Return empty transceiver skip list per DUT. Shared across upgrade-related test suites."""
+    # Required by reboot_and_check(); empty list = no transceivers skipped during post-reboot health check.
     return {dut.hostname: [] for dut in duthosts}
-
 
 TMP_VLAN_PORTCHANNEL_FILE = '/tmp/portchannel_interfaces.json'
 TMP_VLAN_FILE = '/tmp/vlan_interfaces.json'

--- a/tests/common/platform/reboot_utils.py
+++ b/tests/common/platform/reboot_utils.py
@@ -7,6 +7,7 @@ from tests.common.reboot import (
     REBOOT_TYPE_COLD,
     wait_for_startup,
 )
+from tests.common.utilities import wait_until
 
 
 def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list,
@@ -32,9 +33,21 @@ def reboot_and_check(localhost, dut, interfaces, xcvr_skip_list,
     sync_reboot_history_queue_with_dut(dut)
 
     logging.info("Run %s reboot on DUT" % reboot_type)
+    is_dpu_reboot = (invocation_type == "gnoi_based"
+                     and ptf_gnoi is not None
+                     and ptf_gnoi.grpc_client.ss_target_index is not None)
     reboot(dut, localhost, reboot_type=reboot_type,
            reboot_helper=reboot_helper, reboot_kwargs=reboot_kwargs, invocation_type=invocation_type,
-           ptf_gnoi=ptf_gnoi)
+           ptf_gnoi=ptf_gnoi,
+           wait_for_ssh=not is_dpu_reboot)
+
+    # For gNOI DPU reboot the NPU stays up so SSH wait is a no-op.
+    # Poll the DPU's gNOI Time RPC until the DPU server is back.
+    if is_dpu_reboot:
+        dpu_reboot_timeout = 600
+        logging.info("Waiting for DPU gNOI server to come back up (timeout=%ds)", dpu_reboot_timeout)
+        assert wait_until(dpu_reboot_timeout, 10, 0, ptf_gnoi.system_time), \
+            "DPU gNOI server did not come back up within %ds after reboot" % dpu_reboot_timeout
 
     # Append the last reboot type to the queue
     logging.info("Append the latest reboot type to the queue")

--- a/tests/common/ptf_grpc.py
+++ b/tests/common/ptf_grpc.py
@@ -147,6 +147,12 @@ class PtfGrpc:
             for name, value in items:
                 cmd.extend(["-H", f"{name}: {value}"])
 
+        # Auto-inject ss_target headers set via with_ss_target() so callers don't need to pass metadata
+        if self.ss_target_type:
+            cmd.extend(["-H", f"x-sonic-ss-target-type: {self.ss_target_type}"])
+        if self.ss_target_index is not None:
+            cmd.extend(["-H", f"x-sonic-ss-target-index: {self.ss_target_index}"])
+
         # Add custom headers
         for name, value in self.headers.items():
             cmd.extend(["-H", f"{name}: {value}"])

--- a/tests/common/reboot.py
+++ b/tests/common/reboot.py
@@ -390,14 +390,19 @@ def reboot(duthost, localhost, reboot_type='cold', delay=10,
         collect_console_log, args=(duthost, localhost, timeout + wait_conlsole_connection))
     time.sleep(wait_conlsole_connection)
     # Perform reboot
-    if duthost.dut_basic_facts()['ansible_facts']['dut_basic_facts'].get("is_smartswitch"):
+    if duthost.dut_basic_facts()['ansible_facts']['dut_basic_facts'].get("is_smartswitch") \
+            and invocation_type != "gnoi_based":
         reboot_res, dut_datetime = reboot_smartswitch(duthost, pool, reboot_type)
     else:
         reboot_res, dut_datetime = perform_reboot(duthost, pool, reboot_command, reboot_helper,
                                                   reboot_kwargs, reboot_type, invocation_type, localhost,
                                                   ptf_gnoi=ptf_gnoi)
 
-    wait_for_shutdown(duthost, localhost, delay, timeout, reboot_res)
+    is_dpu_reboot = (invocation_type == "gnoi_based"
+                     and ptf_gnoi is not None
+                     and ptf_gnoi.grpc_client.ss_target_index is not None)
+    if not is_dpu_reboot:
+        wait_for_shutdown(duthost, localhost, delay, timeout, reboot_res)
 
     # Release event to proceed poweron for PDU.
     power_on_event.set()

--- a/tests/upgrade_path/test_upgrade_gnoi.py
+++ b/tests/upgrade_path/test_upgrade_gnoi.py
@@ -29,7 +29,7 @@ def gnoi_upgrade_path_lists(request):
 def test_upgrade_via_gnoi(
     localhost, duthosts, ptfhost, rand_one_dut_hostname,
     nbrhosts, fanouthosts, tbinfo, request,
-    gnoi_upgrade_path_lists, gnmi_tls,  # noqa: F811
+    gnoi_upgrade_path_lists, ptf_gnoi,  # noqa: F811
     conn_graph_facts, xcvr_skip_list
 ):
     duthost = duthosts[rand_one_dut_hostname]

--- a/tests/upgrade_path/test_upgrade_smart_switch_gnoi.py
+++ b/tests/upgrade_path/test_upgrade_smart_switch_gnoi.py
@@ -125,6 +125,8 @@ def test_upgrade_one_dpu_via_gnoi(
         tbinfo=tbinfo,
         ptf_gnoi=ptf_gnoi,
         cfg=cfg,
+        localhost=localhost,
+        duthosts=duthosts,
     )
 
 
@@ -168,6 +170,7 @@ def test_upgrade_multiple_dpus_via_gnoi_parallel(
             protocol="HTTP",
             allow_fail=False,
             to_version=to_version,
+            ss_target_index=idx,
             metadata=_build_dpu_metadata(idx),
             ss_reboot_ready_timeout=reboot_ready_timeout,
             ss_reboot_message="Rebooting DPU for maintenance (gNOI upgrade parallel)",
@@ -179,6 +182,8 @@ def test_upgrade_multiple_dpus_via_gnoi_parallel(
         ptf_gnoi=ptf_gnoi,
         cfgs=cfgs,
         max_workers=workers,
+        localhost=localhost,
+        duthosts=duthosts,
     )
 
 

